### PR TITLE
Improve mineral handling

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

File: src/prototype_creep_mineral.js [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+  this.moveToMy(this.room.terminal.pos);
+  for (let transfer of Object.keys(this.carry)) {
+    let resource = this.transfer(this.room.terminal, transfer);
+  }
+};
+
+Creep.prototype.withdrawAllMineralsFromStorage = function() {
+  this.moveToMy(this.room.storage.pos);
+  for (let resource in this.room.storage.store) {
+    if (resource === 'energy' || resource === 'power') {
```
Comment (samogot) [inline src/prototype_creep_mineral.js]: `RESOURCE_`* constants would be better

File: src/prototype_room_mineral.js [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+Room.prototype.isMineralInStorage = function() {
+  let mineralInStorage = false;
+  for (let resource in this.storage.store) {
+    if (resource === 'energy' || resource === 'power') {
+      continue;
+    }
+    mineralInStorage = true;
+    break;
+  }
+  return mineralInStorage;
```
Comment (samogot) [inline src/prototype_room_mineral.js]: Object.keys(this.storage.store).some(r => r !== RESOURCE_ENERGY && r !== RESOURCE_POWER)

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 243/427
    Coefficient: 0.569086651054
    Merging in 3 days 4 hours
    Age 1 days 3 hours

File: src/prototype_creep_mineral.js (hunk 1 above)
Comment (TooAngel) [inline src/prototype_creep_mineral.js]: thx

File: src/prototype_room_mineral.js (hunk 2 above)
Comment (TooAngel) [inline src/prototype_room_mineral.js]: thx, appreciated

**Pull request merged**